ouster-lidar/ouster-ros
# SW-5466: Support Velodyne and other point types in ouster-ros driver

## Related Issues & PRs
- resolves #15 
- resolves #29
- resolves #210 
- related #97
- related #217
- related #223

## Summary of Changes
- added the ability to customize the published point clouds(s) to velodyne point cloud format and
  other common pcl point types.
- made ouster_image_component operate separately from ouster_cloud_component

## Validation
- Run the driver with default parameters and verify no observable behavioral changes.
- Change `point_type` parameter to `native` and verify that it reflects the same point cloud format that we get from the sensor.
- Change `point_type` parameter to `xyz` and verify that the published point cloud format only has `xyz` fields, regardless of the underlying sensor format.
- Change `point_type` parameter to `xyzi` and verify that the published point cloud format only has `xyz` fields plus intensity, regardless of the underlying sensor format (you should get a warning when the low data mode is selected for this mode).
- Change `point_type` parameter to `xyzir` and verify that the published point cloud format only has `xyz` fields plus intensity plus ring, regardless of the underlying sensor format (you should get a warning when the low data mode is selected for this mode).
